source/hic_hal/target_config.h

```diff
@@ -55,7 +55,8 @@ typedef struct target_cfg {
     uint8_t erase_reset;            /*!< Reset after performing an erase */
 } target_cfg_t;
 
-extern const target_cfg_t target_device;
+extern target_cfg_t target_device;
+void set_target_device(uint32_t device);
```

Comment:
This is a common file, but only the nrf51/nrf52 provides set_target_device. Maybe instead of adding a new function here, an extern declaration of target_device_nrf52 could be put in mkit_dk_dongle_nrf5x. That way the function set_target_device could be static and live in mkit_dk_dongle_nrf5x.